src/pages/api/apiUtils/userData.ts

```diff
@@ -222,7 +277,7 @@ export const getPeriodGeoZoneTicketJson = async (
         uuid,
         operatorName: operatorObject?.operator?.operatorPublicName,
         zoneName: fareZoneName,
-        products: productDetailsList,
+        products: productDetailsList || salesOfferPackages,
```

Comment:
Tracking these two variables through the code they look like the should be the same, but it's quite difficult to tell. Is there any way we can make it easier to trace? The `getProducts` function feels like its doing quite a lot and the code above seems to be inversely checking the same parameters as the `getProducts` function.